fluss-lake/fluss-lake-common/src/main/java/com/alibaba/fluss/lake/common/assigner/BucketAssigner.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2024 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.lake.common.assigner;
+
+import com.alibaba.fluss.row.InternalRow;
+
+/** The BucketAssigner interface for assigning buckets to rows. */
+public interface BucketAssigner {
```

Comment:
We have introduced this thing in https://github.com/alibaba/fluss/pull/362. 
But renamed to `BucketingFunction` to avoid naming conflict with existing `BucketAssigner`. So we don't need to introduce `BucketAssigner` and `BucketAssignerFactory` anymore.